torchvision/prototype/datasets/utils/_dataset.py

```diff
@@ -181,3 +181,27 @@ def load(
 
     def _generate_categories(self, root: pathlib.Path) -> Sequence[Union[str, Sequence[str]]]:
         raise NotImplementedError
+
+
+class Dataset2(IterDataPipe[Dict[str, Any]], abc.ABC):
+    def __init__(self, root: Union[str, pathlib.Path], *, skip_integrity_check: bool = False) -> None:
```

Comment:
just wondering if there exist a builtin type that is equivalent to `Union[str, pathlib.Path]` ?